frontend/src/utils/websockets.js

```diff
@@ -43,25 +44,32 @@ export function connectWS(topic, onMessageCallback) {
 
   const connect = () => {
     let reconnectCounter = 0;
+    let badGatewayStatus = false;
     this.sockClient = new Client({
       brokerURL: config.WS_URL,
       debug(strMessage) {
         // console.log('debug: ', strMessage);
         // -- detect reconnect and increment the reconnect counter
         if (strMessage.includes('reconnect')) {
           getUserSession()
-            .then(({ data }) => {
+            .then((userSessionResp) => {
```

Comment:
why not simply `.then({ data, status }) => { .....` ?
(instead of introducing that `userSessionResp`)